MetaMask/metamask-extension
# Fix confirmed transaction time

Fixes: 
 Update with the timestamp fetched from Etherscan api 

**Explanation:**  

1. All transactions (current, pending, confirmes, etc.) are saved to the state.  Each new entry has a timestamp assigned to Date.now and they will be used to create/fill  an array  of history of transactions.
2. This timestamp is the one used when we display the time on activities logs.
3. Meaning, if we closed the browser and re-open 20min later, the time will be set as mentioned in bullet 1
4. All TXs info are fetched from  our etherscan api call. The timestamp this call api is giving us is the time mined and ofc the one displayed on etherscan view

**Manual testing steps:**  
  - Create a transaction
  - Close the browser
  - Wait 2 or 3min
  - Open browser and metamask, log
  - You will see notification about TX confirmation
  - Click  on the TX, and you will see the time of the confirmed transaction is the one from  etherscan view.
  
 **Breaking:**
- The displayed calculated confirmation time is now the time where the transaction was  mined. (from etherscan)
 
  Issue: https://github.com/MetaMask/metamask-extension/issues/7135
